### PR TITLE
feat: support for various canvas sizes

### PR DESCRIPTION
**Pull request by mpryor**
## Goal
The main intent of this PR is to allow the rendering of the graph to better handle different canvas sizes. 

As a side-effect of introducing this code, it seemed necessary to introduce some level of regression testing into the framework for future changes.  

For this reason, I also introduced [Cypress](https://www.cypress.io/) into the `draw` package, along with a [visual regression testing plugin](https://github.com/cypress-visual-regression/cypress-visual-regression). 

The screenshots added in this repo set up a "baseline" that can be compared to during CI in the `visualtesting.cy.js` file.

With the introduction of visual testing, I also added a `randomizeStartingPoints` configuration option to the simulation, which can be used to disable the randomness of the simulation - allowing deterministic results for our visual tests. 

## Examples
Currently, on the main branch, if you setup an index.html with the following code
```
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Mind Graph</title>
  </head>
  <body style="background-color: #001215">
    <canvas
      id="app"
      style="width: 400px; height: 400px; position: absolute; top: 0; left: 0"
    />
    <script type="module" src="/example/main.ts"></script>
  </body>
</html>
```
The result is the following image, where the aspect ratio and sizing of the graph is completely skewed.
image

Using that same HTML file with the code on this branch, the result is the following:
image

### CODE REVIEW

File: packages/draw/src/canvas.ts [hunk 1]
```diff
@@ -5,25 +5,27 @@ import { Zoomer } from './zoomer';
...
+    this.setDimensions();
   }
 
-  public setDimensions(width: number, height: number): void {
-    const appliedWidth = this.deviceScale ? width * this.deviceScale : width;
-    const appliedHeight = this.deviceScale ? height * this.deviceScale : height;
-
-    this.element.attr('width', appliedWidth);
-    this.element.attr('height', appliedHeight);
+  public setDimensions(): void {
```
Comment (jollyjerr) [inline packages/draw/src/canvas.ts:15]: Nice work here 👍🏼

File: packages/draw/cypress/e2e/visualtesting.cy.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+/// <reference types="cypress" />
+
+describe('mindgraph visual tests', () => {
```
Comment (jollyjerr) [inline packages/draw/cypress/e2e/visualtesting.cy.js:3]: Can we add https://www.npmjs.com/package/eslint-plugin-cypress and include these js files in our lint/format checks?

**Review by jollyjerr — APPROVED** [review]
Super happy with these changes - nice work! I'm not a huge fan of e2e because it can get really flaky, but I'm down to roll with it for now and this setup is pretty nice.

Had a nit about getting eslint to work on the testing files

For some reason tests fail locally for me - any idea what could be up? They look great on ci though.

Screenshot 2023-06-08 at 9 13 08 AM

File: packages/draw/cypress/e2e/visualtesting.cy.js (hunk 2 above)
Comment (mpryor) [reply packages/draw/cypress/e2e/visualtesting.cy.js:3]: Yes! Good callout! added in https://github.com/jollyjerr/mindgraph/pull/11/commits/b17c9688d2109b92b65ba1011de96d7e8d3c52a0

**Pull request merged**